tests/core/test_nlg.py

```diff
@@ -328,3 +328,126 @@ def test_nlg_fill_response_quick_replies(
         filled_slots={quick_replies_slot_name: quick_replies_slot_value},
     )
     assert result == {"quick_replies": str(quick_replies_slot_value)}
+
+
+def test_nlg_conditional_response_variations():
+    responses = {
+        "utter_test": [
+            {
+                "text": "Conditional Response A",
+                "condition": [{"type": "slot", "name": "slot test", "value": "A"}],
+                "channel": "os",
+            },
+            {
+                "text": "Conditional app response A",
+                "condition": [{"type": "slot", "name": "slot test", "value": "A"}],
+                "channel": "app",
+            },
+            {
+                "text": "Conditional Response B",
+                "condition": [{"type": "slot", "name": "slot test", "value": "B"}],
+            },
+            {"text": "Default response"},
+        ]
+    }
+    t = TemplatedNaturalLanguageGenerator(responses=responses)
+
+    condition_a = t._matches_filled_slots(
```

Comment:
I'd vote for testing through the public interface which is `TemplatedNaturalLanguageGenerator.generate` instead of private / protetected methods which are too closely tied to the implementation

Reply:
Yup, all testing-related feedback taken onboard - I wrote these phase 1 tests to roughly check the implementation. I'll refactor them: I'll have a think how best to create a tracker with the slot mappings I want in order to use the `generate` method.